metadata_mapper/mappers/validator.py

```diff
@@ -0,0 +1,740 @@
+                                  rikolti_value,
+                                  comp_value),
+            **error_dict
+        }
+
+        self.errors.add(**error)
+
+    def _default_error(self, validation_def: dict[str, Any],
+                       rikolti_value: Any, comparison_value: Any) -> None:
+        return {
+            "key": self.key,
+            "level": "ERROR",
+            "field": validation_def["field"],
+            "description": "Validation failed",
+            "actual": rikolti_value,
+            "expected": comparison_value
+        }
+
+    @staticmethod
+    def _normalize_validator_results(results: Any) -> list[Any]:
+        """
+        Accepts any arbitrary output from a validator function and
+        turns it into a flat list.
+        """
+        if not isinstance(results, list):
+            ret = [results]
+        else:
+            nested_result_lists = list(map(Validator._ensure_is_list, results))
+            ret = itertools.chain(*nested_result_lists)
+
+        return list(filter(None, ret))
+
+    @staticmethod
+    def _ensure_is_list(value: Any) -> list[Any]:
+        return value if isinstance(value, list) else [value]
+
+
+default_validatable_fields: list[dict] = [
+    # Full fidelity fields
+    # Full type and content matches required
+    {
+        "field": "id",
+        "type": str,
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match
+                        ]
+    },
+    {
+        "field": "identifier",
+        "type": Validator.list_of(str),
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match
+                        ]
+    },
+    {
+        "field": "title",
+        "type": Validator.list_of(str),
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match
+                        ]
+    },
+    {
+        "field": "type",
+        "type": Validator.list_of(str),
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match
+                        ]
+    },
+    {
+        "field": "rights",
+        "type": Validator.list_of(str),
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match
+                        ]
+    },
+    {
+        "field": "rights_uri",
+        "type": Validator.list_of(str),
+        "validations": [
+                        Validator.required_field,
+                        Validator.full_match,
+                        lambda d, r, c: isinstance(r, list) and len(r) == 1
+                        ]
+    },
+    # Partial fidelity fields
+    # Content match required; nulls okay
+    {
+        "field": "alternative_title",
+        "type": str,
+        "validations": [Validator.content_match],
+        "level": "WARN"
```

Comment:
It is supposed to be used as a default in `_normalize_validations`. I'll take a look and make sure that's working as intended.

Reply:
Fixed - specifying a `level` at the validation definition will apply that level to all validation failures (unless otherwise specified on an individual validation)